eosio_build.sh

```diff
@@ -30,60 +30,113 @@
+		exit 1
+	fi
+	if [ -f "${PWD}/CMakeCache.txt" ]; then
+		printf "\\n\\tPlease remove file %s/CMakeCache.txt before building EOS.IO.\\n \\tExiting now.\\n\\n" "${PWD}"
+		exit 1
+	fi
+
+   	function usage()
+   	{ 
+		printf "\\tUsage: %s [Build Option -o <Debug|Release|RelWithDebInfo|MinSizeRel>] [CodeCoverage -c ] [Doxygen -d]\\n\\n" "$0" 1>&2
+		exit 1
+	}
+
 	ARCH=$( uname )
+	BUILD_DIR="${PWD}/build"
+	CMAKE_BUILD_TYPE=Release
 	DISK_MIN=20
+	DOXYGEN=false
+	ENABLE_COVERAGE_TESTING=false
+	TEMP_DIR="/tmp"
 	TIME_BEGIN=$( date -u +%s )
-	DOXYGEN=false #set to true to build docs
+	VERSION=1.2
 
 	txtbld=$(tput bold)
 	bldred=${txtbld}$(tput setaf 1)
 	txtrst=$(tput sgr0)
 
+	if [ $# -ne 0 ]; then
+		while getopts ":cdo:" opt; do
+			case "${opt}" in
+				o )
+					options=( "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+					if [[ "${options[*]}" =~ "${OPTARG}" ]]; then
+						CMAKE_BUILD_TYPE="${OPTARG}"
+					else
+						printf "\\n\\tInvalid argument: %s\\n" "${OPTARG}" 1>&2
+						usage
+						exit 1
+					fi
+				;;
+				c )
+					ENABLE_COVERAGE_TESTING=true
+				;;
+				d )
+					DOXYGEN=true
+				;;
+				\? )
+					printf "\\n\\tInvalid Option: %s\\n" "-${OPTARG}" 1>&2
+					usage
+					exit 1
+				;;		
+				: )
+					printf "\\n\\tInvalid Option: %s requires an argument.\\n" "-${OPTARG}" 1>&2
+					usage
+					exit 1
+				;;
+				* )
+					usage
+					exit 1
+				;;
+			esac
+		done
+	fi
+
 	if [ ! -d .git ]; then
-		printf "\n\tThis build script only works with sources cloned from git\n"
-		printf "\tPlease clone a new eos directory with 'git clone https://github.com/EOSIO/eos --recursive'\n"
-		printf "\tSee the wiki for instructions: https://github.com/EOSIO/eos/wiki\n"
+		printf "\\n\\tThis build script only works with sources cloned from git\\n"
+		printf "\\tPlease clone a new eos directory with 'git clone https://github.com/EOSIO/eos --recursive'\\n"
+		printf "\\tSee the wiki for instructions: https://github.com/EOSIO/eos/wiki\\n"
 		exit 1
 	fi
 
-	STALE_SUBMODS=$(( `git submodule status | grep -c "^[+\-]"` ))
+	STALE_SUBMODS=$(( $(git submodule status | grep -c "^[+\-]") ))
 	if [ $STALE_SUBMODS -gt 0 ]; then
-		printf "\ngit submodules are not up to date\n"
-		printf "\tPlease run the command 'git submodule update --init --recursive'\n"
+		printf "\\n\\tgit submodules are not up to date.\\n"
+		printf "\\tPlease run the command 'git submodule update --init --recursive'.\\n"
 		exit 1
 	fi
 
-	printf "\n\tBeginning build version: ${VERSION}\n"
-	printf "\t$( date -u )\n"
-	printf "\tUser: $( whoami )\n"
-	printf "\tgit head id: $( cat .git/refs/heads/master )\n"
-	printf "\tCurrent branch: $( git branch | grep \* )\n"
-	printf "\n\tARCHITECTURE: ${ARCH}\n"
+	printf "\\n\\tBeginning build version: %s\\n" "${VERSION}"
+	printf "\\t%s\\n" "$( date -u )"
+	printf "\\tUser: %s\\n" "$( whoami )"
+	printf "\\tgit head id: %s\\n" "$( cat .git/refs/heads/master )"
+	printf "\\tCurrent branch: %s\\n" "$( git branch | grep \* )"
```

Comment:
Why not to use `git rev-parse --abbrev-ref HEAD`? As per: [How to get the current branch name in Git?
](https://stackoverflow.com/a/12142066/55075)

Reply:
@pacificcode this suggestion looks like a better approach, what do you think?

Reply:
done, PR will be in the morning.